module-system/sov-state/Cargo.toml

```diff
@@ -37,11 +36,37 @@ tempfile = { workspace = true }
 proptest = { workspace = true }
 
 [features]
+default = ["std"]
 arbitrary = [
     "dep:arbitrary",
     "dep:proptest",
-    "dep:proptest-derive"
+    "dep:proptest-derive",
+    "std"
+]
+bench = [
+    "sov-zk-cycle-macros",
+    "risc0-zkvm",
+    "risc0-zkvm-platform",
+    "std",
+]
+native = []
+sov-db = [
+    "dep:sov-db",
+    "std"
+]
+std = [
+    "anyhow/default",
+    "bcs",
+    "borsh/default",
+    "jmt/default",
+    "serde/default",
+    "sov-rollup-interface/default",
+    "sov-first-read-last-write-cache/default",
+    "sync",
+]
+sync = [
+    "borsh/rc",
+    "jmt",
+    "serde/rc",
+    "sov-first-read-last-write-cache",
 ]
```

Comment:
Do we absolutely need all of these 3 new features? Seems like a lot :/

Reply:
We need to decouple `sov-db` from native as it is std-only.

The `sync` is for bare environments without atomic synchronization (might be the case for some hardware wallets). Basically it would require this state without `sync` capabilities, but still have critical components available such as the `Storage` trait definition

Reply:
Just to add to the discussion, we stumbled upon similar issue before:

https://github.com/Sovereign-Labs/sovereign-sdk/issues/219

So maybe we can reconsider split of `sov-state` create into 2